File: LeetCode/1558. Minimum Numbers of Function Calls to Make Target Array/solution.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        ans = mx = 0
        for a in nums:
            cur = 0
            while a:
                ans += (a&1)
                a >>= 1
                cur += (a>0)
            mx = max(mx, cur)
        return ans+mx

class Solution:
    def minOperations(self, A):
        return sum(bin(a).count('1') for a in A) + len(bin(max(A))) - 3        

# class Solution:
#     def minOperations(self, nums: List[int]) -> int:
#         mul, add = 0, 0
#         for cur in nums:
#             cur = bin(cur)[2:]
#             mul = max(mul, len(cur)-1)
#             add += cur.count('1')
#         return mul+add

File: LeetCode/1558. Minimum Numbers of Function Calls to Make Target Array/solution.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int ones = 0, mul = 0;
        for (auto x : nums){
            for (int i = 0; x > 0; i++){
                ones += (x&1);
                x >>= 1;
                mul = max(mul,i);
            }
        }
        return ones+mul;
    }
};